src/services/refactors/addOrRemoveBracesToArrowFunction.ts

```diff
@@ -0,0 +1,109 @@
+    const refactorName = "Add or remove braces in an arrow function";
+    const refactorDescription = Diagnostics.Add_or_remove_braces_in_an_arrow_function.message;
+    const addBracesActionName = "Add braces to arrow function";
+    const removeBracesActionName = "Remove braces from arrow function";
+    const addBracesActionDescription = Diagnostics.Add_braces_to_arrow_function.message;
+    const removeBracesActionDescription = Diagnostics.Remove_braces_from_arrow_function.message;
+    registerRefactor(refactorName, { getEditsForAction, getAvailableActions });
+
+    interface Info {
+        container: ArrowFunction;
+        expression: Expression;
+        addBraces: boolean;
+    }
+
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+        const { file, startPosition } = context;
+        const info = getConvertibleArrowFunctionAtPosition(file, startPosition);
+        if (!info) return undefined;
+
+        return [{
+            name: refactorName,
+            description: refactorDescription,
+            actions: [
+                info.addBraces ?
+                    {
+                        name: addBracesActionName,
+                        description: addBracesActionDescription
+                    } : {
+                        name: removeBracesActionName,
+                        description: removeBracesActionDescription
+                    }
+            ]
+        }];
+    }
+
+    function getEditsForAction(context: RefactorContext, _actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+        const info = getConvertibleArrowFunctionAtPosition(file, startPosition);
+        if (!info) return undefined;
+
+        const { expression, container } = info;
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+
+        if (_actionName === addBracesActionName) {
+            addBraces(changeTracker, file, container, expression);
+        }
+        else if (_actionName === removeBracesActionName) {
+            removeBraces(changeTracker, file, container, expression);
+        }
+        else {
+            Debug.fail("invalid action");
+        }
+
+        return {
+            renameFilename: undefined,
+            renameLocation: undefined,
+            edits: changeTracker.getChanges()
+        };
+    }
+
+    function addBraces(changeTracker: textChanges.ChangeTracker, file: SourceFile, container: ArrowFunction, expression: Expression) {
+        updateBraces(changeTracker, file, container, createBlock([createReturn(expression)]));
+    }
+
+    function removeBraces(changeTracker: textChanges.ChangeTracker, file: SourceFile, container: ArrowFunction, expression: Expression) {
+        if (!isLiteralExpression(expression) && !isIdentifier(expression) && !isParenthesizedExpression(expression) && expression.kind !== SyntaxKind.NullKeyword) {

```

Comment:
Should have a `needsParentheses` utility function that uses a `switch` statement. There are probably some unhandled cases here -- a property access shouldn't need parentheses, nor should a function call. Nor should `true` or `false` or `undefined`.
(By the way, I don't think it would be worth it to add a test for every single one of these cases. I notice there are a lot of tests related to parenthesizing already -- maybe they should be combined into one file.)